samples/msal-react-native-android-poc-sample/App.js

```diff
@@ -0,0 +1,306 @@
+  StatusBar,
+  TextInput,
+  Button,
+} from 'react-native';
+
+import MSAL from '@azuread/msal-react-native-android-poc';
+
+import ToolbarAndroid from '@react-native-community/toolbar-android';
+
+import {
+  Text,
+} from '@fluentui/react-native';
+
+import {
+  Colors,
+} from 'react-native/Libraries/NewAppScreen';
+
+
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isSharedDevice: false,
+      scopesValue: "",
+      URLValue: "https://graph.microsoft.com/v1.0/me",
+      account: null,
+      graphResult: null
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      await this.getAccount();
+      await this.isSharedDevice();
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  getAccount = async () => {
+    try{
+      const result = await MSAL.getAccount();
+      this.setState({
+        account: result,
+      });
+    } catch (message) {
+      console.log(message);
+    }
+  }
+
+  isSharedDevice = async () => {
+    try {
+      const sharedDevice = MSAL.isSharedDevice();
+      this.setState({
+        isSharedDevice: sharedDevice
+      })
+    } catch (message) {
+      console.log(message);
+    }
+  }
+
+  startSignIn = async () => {
+    try{
+      const result = await MSAL.signIn(this.state.scopesValue);
+      this.setState({
+        account: result,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  startSignOut = async () => {
+    try {
+      const result = await MSAL.signOut();
+      this.setState({
+        account: null,
+        graphResult: null,
+      });
+    } catch (exception) {
+      console.log(exception)
+    }
+  }
+
+  startGetGraphDataSilently = async () => {
+    try {
+      const result = await MSAL.acquireTokenSilent(this.state.scopesValue);
+      //api call to MSGraph
+      const response = await fetch (
+        this.state.URLValue, {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${result.accessToken}`
+          }
+        }
+      );
+      const json = await response.json();
+      this.setState({
+        graphResult: json
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  startGetGraphDataInteractively = async () => {
+    try {
+      const result = await MSAL.acquireToken(this.state.scopesValue);
+      //make api call
+      const response = await fetch (
+        this.state.URLValue, {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${result.accessToken}`
+          }
+        }
+      );
+      const json = await response.json();
+      //update current account and state
+      await this.getAccount();
+      this.setState({
+        graphResult: json,
+      });
+    } catch (exception) {
+      console.log(exception);
+    }
+  }
+
+  showResult = () => {
+   return this.state.graphResult && JSON.stringify(this.state.graphResult, null, 4);
+  }
+  
+  render() {
+    return (
```

Comment:
The render method is a little on the long side. It's definitely not a priority but a possible improvement would be to refactor some of the UI portions out into different components. For example, putting the `View` containing all the action (i.e. Get Graph Data Silently) buttons in a separate component (with it's own stylesheet) could drastically improve the readability of the App component.

Reply:
splitting up the render method makes sense, but while trying to create a new component for the buttons, i've ran into some questions.
The buttons and other objects in my app all depend and modify state variables, so splitting the buttons off as a separate component would mean that they wouldn't have access to modifying and getting updates from the state variables without creating more functions that "lift" the state up (and bring the state down). 
I was wondering what your thoughts were about these potential trade offs... for sure the current render function is way too long and would be improved stylistically with separate components, but I wonder if the added functions and logic with lifting state would also make my relatively small app complicated to read as well.